event/src/main/java/com/bazaarvoice/emodb/event/db/astyanax/ChannelAllocationState.java

```diff
@@ -74,16 +74,26 @@ public synchronized SlabAllocation allocate(PeekingIterator<Integer> eventSizes)
         _slabConsumed += countAndBytesConsumed.getLeft();
         _slabBytesConsumed += countAndBytesConsumed.getRight();
 
-        if (countAndBytesConsumed.getLeft() < remaining) {
+        // Check for case where no more slots could be allocated because slab is full either because
+        // the max # of slots is consumed or the max # of bytes is consumed
+        if (countAndBytesConsumed.getLeft() == 0) {
 
-            // All events fit in current slab, leave it attached
-            return new DefaultSlabAllocation(_slab.addRef(), offsetForNewAllocation, countAndBytesConsumed.getLeft());
+            detach().release();
+            return null;
 
         } else {
```

Comment:
There is no need for an `else`, as we are returning if the above statement is true.